packages/swingset-liveslots/src/watchedPromises.js

```diff
@@ -235,13 +235,13 @@ export function makeWatchedPromiseManager({
       } else {
         watchedPromiseTable.init(vpid, harden([[watcher, ...args]]));
 
+        promiseRegistrations.init(vpid, p);
+        pseudoThen(p, vpid);
```

Comment:
Nit: let's add a comment pointing to the issue for why this must be before the `maybeExportPromise` call.